locobot/agent/perception/handlers/core.py

```diff
@@ -72,17 +79,24 @@ class RGBDepth:
     rgb: np.array
     depth: np.array
     ptcloud: list
-    gray: np.array
 
     def __init__(self, rgb, depth, pts):
         self.rgb = rgb
         self.depth = depth
         self.ptcloud = pts
-        self.gray = cv2.cvtColor(rgb, cv2.COLOR_BGR2GRAY)
 
     def get_pillow_image(self):
         return Image.fromarray(self.rgb, "RGB")
 
+    def get_bounds_for_mask(self, mask):
+        """for all points in the mask, returns the bounds as an axis-aligned bounding box.
+        """
+        indices = zip(*np.where(mask == True))
+        points = [self.get_coords_for_point((x,y)) for x,y in indices]
```

Comment:
these two lines here are obviously super inefficient. do something about it or it'll be a perf bottleneck.

Reply:
vectorized this, roughly 35x faster now (python snippet below)

<img src="https://user-images.githubusercontent.com/57542204/116987363-fa59d900-ac9c-11eb-81b4-ba1987cf956b.png" width="450" height="300">